src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -422,6 +427,37 @@ def generate_request_payload(configuration, location, public_key, tags):
     return cc
 
 
+def check_aks_cluster(kube_config, kube_context):
+    args = ['kubectl', 'cluster-info']
+    if kube_config:
+        args += ["--kubeconfig", kube_config]
+    if kube_context:
+        args += ["--context", kube_context]
+
+    try:
+        proc = run(args, stdout=PIPE, stderr=STDOUT, universal_newlines=True)
```

Comment:
Connectivity check to cluster uses a [different call to the cluster](https://github.com/Azure/azure-cli-extensions/blob/6a96d8bd9c0757612ab069a536116d0a225cfc97/src/connectedk8s/azext_connectedk8s/custom.py#L282) to verify connectivity. Can we update that method to use kubectl cluster-info (using kubernetes client). We can parse output there itself, check if it matches AKS pattern, and print warning. That way we make 1 round trip to cluster instead of 2.

Reply:
Okay, will look into it. Thanks.

Reply:
If we are looking for the corresponding server address, I was thinking to fetch that from kubeconfig itself.

Reply:
Added the approach to fetch server from kubeconfig itself.